### PR TITLE
test: introduce TEST-81-GENERATORS 

### PR DESCRIPTION
**Pull request by mrc0mmand**
Add some explicit tests for various generators we ship, e.g.:
   - systemd-debug-generator
   - systemd-environment-d-generator
   - systemd-fstab-generator

\+ a couple of tests for systemd-modules-load

I originally wanted to submit this once I cover all our generators but I underestimated the amount of options systemd-fstab-generator supports, which made the test *slightly* more complicated. Let's see if CIs like it.

### CODE REVIEW

File: src/fstab-generator/fstab-generator.c [hunk 1]
```diff
@@ -698,10 +698,12 @@ static int parse_fstab(bool initrd) {
                         }
 
                         if (sysfs_check < 0) {
-                                r = getenv_bool_secure("SYSTEMD_SYSFS_CHECK");
-                                if (r < 0 && r != -ENXIO)
-                                        log_debug_errno(r, "Failed to parse $SYSTEMD_SYSFS_CHECK, ignoring: %m");
-                                sysfs_check = r != 0;
+                                int ret;
```
Comment (bluca) [inline src/fstab-generator/fstab-generator.c]: we usually use `k` for this

**Review by bluca — APPROVED** [review]

File: test/units/testsuite-81.debug-generator.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# shellcheck disable=SC2235
+set -eux
+set -o pipefail
+
+# shellcheck source=test/units/assert.sh
```
Comment (yuwata) [inline test/units/testsuite-81.debug-generator.sh]: Maybe, copy and paste error?

File: test/units/testsuite-74.modules-load.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+# Command line arguments
+modprobe -v --all --remove loop dummy
+# Make sure we have no config files left over that might interfere with
+# following tests
+rm -fv "$CONFIG_FILE"
+[[ -z "$(systemd-analyze cat-config modules-load.d)" ]]
+# Copy over the existing kernel cmdline, so we can amend it, and then overmount
+# the original one with the amended cmdline
+cp -fv /proc/cmdline /tmp/cmdline
+sed -i 's/$/ modules_load= modules_load=, modules_load=foo-bar-baz,dummy modules_load=loop,loop,loop/' /tmp/cmdline
```
Comment (yuwata) [inline test/units/testsuite-74.modules-load.sh]: Not a matter, but using `SYSTEMD_PROC_CMDLINE` is easier?

File: test/units/testsuite-81.debug-generator.sh [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+# Initrd scenario
+: "debug-shell: initrd"
+CMDLINE="ro root=/ ${ARGS[*]} systemd.debug_shell"
+SYSTEMD_IN_INITRD=1 SYSTEMD_PROC_CMDLINE="$CMDLINE" run_and_list "$GENERATOR_BIN" "$OUT_DIR"
+link_eq "$OUT_DIR/masked-initrd.service" /dev/null
+link_endswith "$OUT_DIR/initrd.target.wants/wanted-initrd.service" /lib/systemd/system/wanted-initrd.service
+# The non-initrd stuff (i.e. without the rd. suffix) should be ignored in
+# this case
+(! test -h "$OUT_DIR/masked-no-suffix.service")
```
Comment (yuwata) [inline test/units/testsuite-81.debug-generator.sh]: Why not to use `test ! -e ...`?

File: src/fstab-generator/fstab-generator.c (hunk 1 above)
Comment (mrc0mmand) [reply src/fstab-generator/fstab-generator.c]: Ah, indeed, thanks.

File: test/units/testsuite-81.debug-generator.sh (hunk 2 above)
Comment (mrc0mmand) [reply test/units/testsuite-81.debug-generator.sh]: Whoops, definitely.

File: test/units/testsuite-74.modules-load.sh (hunk 3 above)
Comment (mrc0mmand) [reply test/units/testsuite-74.modules-load.sh]: Ah, yeah, I realized that when writing fstab-generator tests but forgot to convert this one, thanks.

File: test/units/testsuite-81.debug-generator.sh (hunk 4 above)
Comment (mrc0mmand) [reply test/units/testsuite-81.debug-generator.sh]: No idea why I didn't think of that :)

**Review by yuwata — APPROVED** [review]

**Pull request merged**